core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -733,11 +756,22 @@ trait WorkspaceComponent {
   }
 
   private def groupByWorkspaceId(runningSubmissions: Seq[(UUID, Int)]): Map[UUID, Int] = {
-    runningSubmissions.groupBy{ case (wsId, count) => wsId }.mapValues { case Seq((_, count)) => count }
+    CollectionUtils.groupPairs(runningSubmissions.toList)
   }
 
   private def groupByWorkspaceIdThenStatus(workflowDates: Seq[(UUID, String, Option[Timestamp])]): Map[UUID, Map[String, Option[Timestamp]]] = {
-    workflowDates.groupBy { case (wsId, _, _) => wsId }.mapValues(_.groupBy { case (_, status, _) => status }.mapValues { case Seq((_, _, timestamp)) => timestamp })
+    // There is no Monoid instance for Option[Timestamp] so we need to bring one into scope.
+    // However a Monoid for Timestamp doesn't really make sense -- what would it do, add them together?
+    // We can take advantage of the _universal_ monoid for Option which combines Option values using
+    // Option.orElse. It's called universal because it works no matter the type inside the Option.
+    // This is fine in this case because there are guaranteed no key conflicts due to the SQL query
+    // structure (group by, etc).
+    //
+    // TL/DR: The following line brings into scope a Monoid[Option[Timestamp]] which combines values
+    // using Option.orElse.
```

Comment:
I think part of my confusion here is that you're introducing Monoid, which is used to combine things, using a non-combining operation (`orElse`), in a situation where you never need to combine two of them anyway because the UUID/String combo is guaranteed to be unique. This is a pretty head-bendy way to achieve the desired result, even if it does work.